interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java

```diff
@@ -337,35 +372,16 @@ public void sendHeaders(Metadata responseHeaders) {
                 new Metadata());
             return noopListener;
           }
-          break;
         }
-      }
-      int attemptNum = 0;
-      String attemptNumHeader = requestHeaders.get(ATTEMPT_NUM);
-      if (attemptNumHeader != null) {
-        try {
-          attemptNum = Integer.valueOf(attemptNumHeader);
-        } catch (NumberFormatException e) {
-          newCall.close(
-              Status.INVALID_ARGUMENT.withDescription(
-                  String.format(
-                      "Invalid format for grpc-previous-rpc-attempts header (%s)",
-                      attemptNumHeader)),
-              new Metadata());
-          return noopListener;
+        if (attemptNum == succeedOnAttemptNum) {
```

Comment:
Seems we can move this into the `callBehavior.startsWith(CALL_BEHAVIOR_SUCCEED_ON_RETRY_ATTEMPT_VALUE)` block now.

Reply:
Done.